appcues/src/main/java/com/appcues/ui/primitive/OptionSelectPrimitive.kt

```diff
@@ -55,14 +52,11 @@ import com.appcues.ui.extensions.styleBorder
 @Composable
 internal fun OptionSelectPrimitive.Compose(modifier: Modifier) {
     val formState = LocalExperienceStepFormStateDelegate.current
-    val selectedValues = remember { mutableStateOf(defaultValue) }
+    val selectedValues = remember { mutableStateOf(formState.getValue(this)) }
 
-    LaunchedEffect(key1 = selectedValues.value) {
-        val formItem = when (selectMode) {
-            MULTIPLE -> MultipleTextFormItem(label.text, required, selectedValues.value)
-            SINGLE -> SingleTextFormItem(label.text, required, selectedValues.value.firstOrNull() ?: "")
-        }
-        formState.captureFormItem(this@Compose.id, formItem)
+    fun setSelectedValues(values: Set<String>) {
```

Comment:
I'm not sure if its Ok to declare a function inside a composition

Reply:
no strong opinion, I was wondering what happens when during the runtime the composition reads a function there, but maybe this is nothing to be concerned about, probably same thing happens with inline functions no that I think abou it